### ISSUE TITLE
Questions about transactions

### ISSUE THREAD

**Issue by RamiroMoreira** [input-output-hk/cardano-sl#2381]
I have been tring to ask a cardano developer about the way the transactions works. Don't know if this is the place to ask. I'm integrating cardano to an app, and i run into this issue:

I have a cardano wallet, the wallet has an account "Ac1", and the account has an address "Ad1"
Ad1 has X ADA.

Then i use: https://cardanodocs.com/technical/wallet/api/#/default/post_api_txs_payments__from___to___amount_
to send Y ADA from "Ac1" to an address "Ad2" outside my wallet (from another person or user).

So i explore my transaction and it has one "from" and two destination, it sended Y ADA to "Ad2" and X-Y ADA to a new address "Ad3" inside the account "Ac1".
In summary... every time i send a transaction, cardano send my remaining balance to a new address in my account. I read about it and is a bitcoin standard, it's for security? is there a way to make the remaining balance to stay in the same address? should i continue using the original address? 

Sorry for the bad explanation. Thanks

**Comment by hithroc** [comment]
Yes, this is done for security. Whenever you make a transaction you sign it, and that exposes your public key, and since there exists a quantum integer factorization algorithm, we'd rather not have public key exposed as much as possible. That's why the default grouping policy for transactions is to send all reamining balance to a new address for which the public key is not yet exposed.

Since #1826 we have two grouping policies, but I'm not sure how to specify them in the API, I think there are people who know that better than me.

**Comment by RamiroMoreira** [comment]
oh, that should be: 
"InputSelectionPolicystring
Enum:
[ OptimizeForSecurity, OptimizeForSize ]" 
but i don't know what they mean since they are not explained on the documentation. Thanks for anwser.

**Comment by Martoon-00** [comment]
Thanks for your question!

In addition to security concern, also privacy comes into play, because using same address for various purposes may potentially reveal all your transactions to one of the parties involved.

Way to send change not to another address is not supported. And this seems to be restricting in no way, because whenever one wants to spend money, he has to specify account as a source, not an address. Addresses are throwaway stuff, and any single address shouldn't be used twice.

Mentioned grouping policy for transactions allows to specify, whether is it allowed to spend money of address partially if it occurred to be possible due to inner mechanism, or addresses should be always spent fully (also for security reasons) - and the latter behaviour is default one.
